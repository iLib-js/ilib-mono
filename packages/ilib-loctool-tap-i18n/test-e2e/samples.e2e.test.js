/*
 * samples.e2e.test.js - E2E tests for ilib-loctool-tap-i18n samples
 *
 * Copyright Â© 2025, Box, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

const path = require("path");
const fs = require("fs");
const { expectFileToMatchSnapshot, LoctoolRunner } = require("ilib-internal");

describe("samples", () => {
    describe("tap-i18n", () => {
        const projectPath = path.resolve(__dirname, "..", "samples", "tap-i18n");
        const xliffPath = path.resolve(projectPath, "sample-tap-i18n-extracted.xliff");

        beforeAll(async () => {
            const loctool = new LoctoolRunner(projectPath);
            await loctool.run("localize");
        });

        afterAll(() => {
            // Clean up all generated files and directories
            const filesToClean = [
                xliffPath, // sample-tap-i18n-extracted.xliff
                path.resolve(projectPath, "sample-tap-i18n-new-ko-KR.xliff"),
                path.resolve(projectPath, "sample-tap-i18n-new-nl-NL.xliff"),
                path.resolve(projectPath, "languages/ko-KR.i18n.yml"),
                path.resolve(projectPath, "languages/nl-NL.i18n.yml")
            ];
            
            filesToClean.forEach(file => {
                if (fs.existsSync(file)) {
                    if (fs.statSync(file).isDirectory()) {
                        fs.rmSync(file, { recursive: true, force: true });
                    } else {
                        fs.unlinkSync(file);
                    }
                }
            });
        });

        it("should produce an extracted XLIFF file", () => {
            expectFileToMatchSnapshot(xliffPath);
        });
    });
});
