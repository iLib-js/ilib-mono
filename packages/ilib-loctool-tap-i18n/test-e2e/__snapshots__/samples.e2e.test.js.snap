// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`samples tap-i18n should produce an extracted XLIFF file 1`] = `
"<?xml version="1.0" encoding="utf-8"?>
<xliff version="1.2">
  <file original="languages/en-US.i18n.yml" source-language="en-US" product-name="sample-tap-i18n">
    <body>
      <trans-unit id="1" resname="toplevel.plurals.plural1" restype="plural" datatype="x-yaml" extype="one">
        <source>{{count}} item</source>
        <note>This is the comment for the first plural string</note>
      </trans-unit>
      <trans-unit id="2" resname="toplevel.plurals.plural1" restype="plural" datatype="x-yaml" extype="other">
        <source>{{count}} items</source>
        <note>This is the comment for the first plural string</note>
      </trans-unit>
      <trans-unit id="3" resname="toplevel.first.loading" restype="string" datatype="x-yaml">
        <source>Loading...</source>
        <note>This is the comment for the first string</note>
      </trans-unit>
      <trans-unit id="4" resname="toplevel.first.pageLoading" restype="string" datatype="x-yaml">
        <source>Loading the page...</source>
        <note>This is the comment for the second string</note>
      </trans-unit>
      <trans-unit id="5" resname="toplevel.first.saveChanges" restype="string" datatype="x-yaml">
        <source>Save Changes</source>
        <note>Label for the button that saves all changes for the search configuration.</note>
      </trans-unit>
      <trans-unit id="6" resname="toplevel.second.cancel" restype="string" datatype="x-yaml">
        <source>Cancel</source>
        <note>Label for the button that cancels an action.</note>
      </trans-unit>
      <trans-unit id="7" resname="toplevel.second.create" restype="string" datatype="x-yaml">
        <source>Create</source>
        <note>Label for the button creating a new item.</note>
      </trans-unit>
    </body>
  </file>
</xliff>"
`;
